hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/checksum/ContainerChecksumTreeManager.java

```diff
@@ -133,42 +167,47 @@ private Lock getWriteLock(long containerID) {
     return fileLock.get(containerID).writeLock();
   }
 
-  private ContainerProtos.ContainerChecksumInfo read(ContainerData data) throws IOException {
+  private Optional<ContainerProtos.ContainerChecksumInfo.Builder> read(ContainerData data) throws IOException {
     long containerID = data.getContainerID();
+    File checksumFile = getContainerChecksumFile(data);
     Lock readLock = getReadLock(containerID);
     readLock.lock();
```

Comment:
Since the code does rename now, there should not be a need for a read lock. The only reason we need write lock is as we do a ReadModifyWrite cycle which need protection between writes but readers are dependent on atomic rename.

Reply:
Makes sense, I'll update this in the next commit.